### ISSUE TITLE
values in `/proc/meminfo` turns filled with zero, it breaks the world

### ISSUE THREAD

**Issue by iav** [koverstreet/bcachefs#447]
Kernel 5.19.y with bcachefs have data in `/proc/meminfo` filled with zero.
example with bcachefs patches:
```
cat /proc/meminfo
MemTotal:       03944876 kB
MemFree:        03484312 kB
MemAvailable:   03674800 kB
Buffers:        00005884 kB
Cached:         00237676 kB
SwapCached:     00000000 kB
Active:         00135704 kB
Inactive:       00160948 kB
Active(anon):   00000740 kB
Inactive(anon): 00058304 kB
Active(file):   00134964 kB
Inactive(file): 00102644 kB
Unevictable:    00000000 kB
Mlocked:        00000000 kB
SwapTotal:      01972436 kB
SwapFree:       01972436 kB
Zswap:                 0 kB
Zswapped:              0 kB
Dirty:          00001156 kB
Writeback:      00000000 kB
AnonPages:      00053140 kB
Mapped:         00043528 kB
Shmem:          00005904 kB
KReclaimable:   00029660 kB
Slab:           00072780 kB
SReclaimable:   00029660 kB
SUnreclaim:     00043120 kB
KernelStack:        3136 kB
PageTables:     00001868 kB
NFS_Unstable:   00000000 kB
Bounce:         00000000 kB
WritebackTmp:   00000000 kB
CommitLimit:    03944872 kB
Committed_AS:   00166948 kB
VmallocTotal:   133143592960 kB
VmallocUsed:    00048384 kB
VmallocChunk:   00000000 kB
Percpu:         00003096 kB
AnonHugePages:  00000000 kB
ShmemHugePages: 00000000 kB
ShmemPmdMapped: 00000000 kB
FileHugePages:  00000000 kB
FilePmdMapped:  00000000 kB
CmaTotal:       00131072 kB
CmaFree:        00122864 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:               0 kB
```
It breaks much programs that read data from `/proc/meminfo` and use internal parsers of some libs, who treat `0010` like octal value, then 0010 will be 8, not 10, as it should. One of sample is `bash`.
Other sample of broken things: https://github.com/Frogging-Family/linux-tkg/issues/550

I think it is a bug, data format can't be "just changed" such way.

**Comment by koverstreet** [comment]
Why are you reporting this here? This isn't a bcachefs bug; this bug tracker is for bcachefs and associated work.

**Comment by iav** [comment]
I think bcachefs did it. 
Because it appears after applying bcachefs patches over kernel code.
Am I wrong, and bcachefs not does that?

**Comment by firestack** [comment]
Linux tkg has had reported issues and has disabled bcachefs to fix it @koverstreet.

https://github.com/Frogging-Family/linux-tkg/issues/550

**Comment by koverstreet** [comment]
I thought I was checking the behaviour of stock v5.19 and apparently I wasn't

**Comment by koverstreet** [comment]
And thanks for the report - this was a printbuf regression

**Comment by koverstreet** [comment]
Fixed by 0d2b0bf193

**Comment by iav** [comment]
confirm fixed for me. 
Thank you!
